crates/next-core/src/next_font_google/util.rs

```diff
@@ -0,0 +1,550 @@
+use std::cmp::Ordering;
+
+use anyhow::{anyhow, Context, Result};
+use indexmap::{indexset, IndexSet};
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::options::{FontData, FontWeights};
+
+const GOOGLE_FONTS_STYLESHEET_URL: &str = "https://fonts.googleapis.com/css2";
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct FontAxes {
+    pub(crate) wght: IndexSet<String>,
+    pub(crate) ital: IndexSet<FontItal>,
+    pub(crate) variable_axes: Option<Vec<(String, String)>>,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub(crate) enum FontItal {
+    Italic,
+    Normal,
+}
+
+// Derived from https://github.com/vercel/next.js/blob/9e098da0915a2a4581bebe2270953a1216be1ba4/packages/font/src/google/utils.ts#L232
+pub(crate) fn get_font_axes(
+    font_data: &FontData,
+    font_family: &str,
+    weights: &FontWeights,
+    styles: &IndexSet<String>,
+    selected_variable_axes: &Option<Vec<String>>,
+) -> Result<FontAxes> {
+    let all_axes = &font_data
+        .get(font_family)
+        .context("Font family not found")?
+        .axes;
+
+    let Some(defineable_axes) = all_axes else {
+        return Err(anyhow!("Font {} has no definable `axes`", font_family));
```

Comment:
Nit: this is just [bail!](https://docs.rs/anyhow/latest/anyhow/macro.bail.html)
```suggestion
        return bail!("Font {} has no definable `axes`", font_family);
```